### PR TITLE
Upgrade `travis.yml` to use `haskell-ci == 0.10.1`

### PR DESCRIPTION
**Pull request by emilypi**
Small maintenance bump for the `.travis.yml`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,14 @@
 # This Travis job script has been generated by a script via
 #
-#   haskell-ci '--benchmarks-jobs=>= 7.10' '--ghc-head' '--installed=-containers' '--installed=-binary' 'unordered-containers.cabal'
+#   haskell-ci '--benchmarks-jobs= >=7.10' '--ghc-head' '--installed=-containers' '--installed=-binary' 'unordered-containers.cabal'
+#
+# To regenerate the script (for example after adjusting tested-with) run
+#
+#   haskell-ci regenerate
```
Comment (sjakobi) [inline .travis.yml:7]: Neat! It's good to have that info right in the file header.

**Comment by emilypi** [comment]
@sjakobi done 😄

**Comment by emilypi** [comment]
Note: this will subsume #218

File: .travis.yml [hunk 2]
```diff
@@ -29,26 +33,29 @@ before_cache:
   - rm -rfv $CABALHOME/packages/head.hackage
 jobs:
   include:
-    - compiler: ghc-8.8.1
-      addons: {"apt":{"sources":[{"sourceline":"deb http://ppa.launchpad.net/hvr/ghc/ubuntu xenial main","key_url":"https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x063dab2bdc0b3f9fcebc378bff3aeacef6f88286"}],"packages":["ghc-8.8.1","cabal-install-3.0"]}}
+    - compiler: ghc-8.10.1
+      addons: {"apt":{"sources":[{"sourceline":"deb http://ppa.launchpad.net/hvr/ghc/ubuntu xenial main","key_url":"https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x063dab2bdc0b3f9fcebc378bff3aeacef6f88286"}],"packages":["ghc-8.10.1","cabal-install-3.2"]}}
+      os: linux
+    - compiler: ghc-8.8.3
```
Comment (emilypi) [inline .travis.yml:39]: NB: bumps GHC 8.8 version to latest stable GHC 8.8.3

**Pull request merged**